### PR TITLE
Create swap.cpp

### PR DESCRIPTION
**Pull request by helper-uttam**
This program uses XOR bitwise and swap the values of 2 variables

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
This program uses XOR bitwise and swap 2 values.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591"></a>

### CODE REVIEW

**Review by mishraabhinn — CHANGES_REQUESTED** [review]
@helper-uttam Proper Documentation and formatting needed. You can also have a look for previous successfully pr merged for reference. Thanks. 
[Refer](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md)

**Comment by helper-uttam** [comment]
> @helper-uttam Proper Documentation and formatting needed. You can also have a look for previous successfully pr merged for reference. Thanks.
> [Refer](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md)

I had added documentation related with every line of code. Now if there anything left please let me know?

File: bit_manipulation/swap.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ //Printing Values of A and B before swapping.
+  cout<<"The Value of A before swaping is = "<<a<<"\n";
+  cout<<"The Value of B before swaing is = "<<b<<"\n;
+  
+    swap_XOR(ptr1,ptr2);    /* here we are passing two pointers which is also known as pass by reference in which we 
+  pass address of variable instead of the value*/
+ 
+ //Printing Values of A and B after swapping.
+   cout<<"The Value of A after swaping is = "<<a<<"\n";
+   cout<<"The Value of B after swaing is = "<<b<<"\n;
```
Comment (mishraabhinn) [inline bit_manipulation/swap.cpp:53]: This should be separated out in a void test type function and using assert you should verify the test cases.

**Review by mishraabhinn — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thank you for submitting a PR! 😄 Sadly, this does not meet our requirements. Check out our [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Please read them **carefully** and **follow them**. Also, check comments by @mishraabhinn. 🙂

**Comment by helper-uttam** [comment]
can someone please suggest me some changes. Like `assert(bit_manipulation::swap::swap_XOR(11, 2) == (2,11));` this require `swap_XOR` to return 2values but Is it possible to return two values from a function?

**Comment by mishraabhinn** [comment]
> can someone please suggest me some changes. Like `assert(bit_manipulation::swap::swap_XOR(11, 2) == (2,11));` this require `swap_XOR` to return 2values but Is it possible to return two values from a function?

Passs referrece in the swap function insted of pointer and compare individual values  using assert.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591/checks?check_run_id=3526813195) warnings.

**Comment by helper-uttam** [comment]
> Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591/checks?check_run_id=3526813195) warnings.

Now I'm not getting the error, This time it occured because pointer's are not initialized but I had initialized them in the next line.

**Comment by Panquesito7** [comment]
> > Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591/checks?check_run_id=3526813195) warnings.
> 
> Now I'm not getting the error, This time it occured because pointer's are not initialized but I had initialized them in the next line.

I still see the error here. Could you please fix it? Thanks.

**Comment by helper-uttam** [comment]
> > > Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591/checks?check_run_id=3526813195) warnings.
> > 
> > 
> > Now I'm not getting the error, This time it occured because pointer's are not initialized but I had initialized them in the next line.
> 
> I still see the error here. Could you please fix it? Thanks.

I don't know how to fix this error :( Sorry

**Comment by mishraabhinn** [comment]
[Refer](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591/checks?check_run_id=3548622422) @helper-uttam 
Screenshot 2021-09-09 at 00 49 13
You can rectify these small errors.

**Comment by helper-uttam** [comment]
> [Refer](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591/checks?check_run_id=3548622422) @helper-uttam
> Screenshot 2021-09-09 at 00 49 13
> You can rectify these small errors.

committed changes again, let's hope for the best :)

**Comment by mishraabhinn** [comment]
> > [Refer](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1591/checks?check_run_id=3548622422) @helper-uttam
> > Screenshot 2021-09-09 at 00 49 13
> > You can rectify these small errors.
> 
> committed changes again, let's hope for the best :)
Great automated test passing. Now i want you to make main() function free of any code and extract all the test from there to test function and verify using assert.

File: bit_manipulation/swap.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ //Printing Values of A and B before swapping.
+  std::cout<<"The Value of A before swaping is = "<<a<<"\n";
+  std::cout<<"The Value of B before swaing is = "<<b<<"\n";
+  
+    bit_manipulation::swap::swap_XOR(ptr1,ptr2);    /* here we are passing two pointers which is also known as pass by reference in which we 
+  pass address of variable instead of the value*/
+ 
+ //Printing Values of A and B after swapping.
+   std::cout<<"The Value of A after swaping is = "<<a<<"\n";
+   std::cout<<"The Value of B after swaping is = "<<b<<"\n";
```
Comment (mishraabhinn) [inline bit_manipulation/swap.cpp:65]: Extract this to test and verify using assert

Comment (helper-uttam) [reply bit_manipulation/swap.cpp]: done!

**Pull request closed without merging**